File: python/bob/bob.py
def response(hey_bob):
    if hey_bob.strip() == '':
        return "Fine. Be that way!"
    elif hey_bob.upper() == hey_bob and ''.join(x.upper() for x in hey_bob if x.upper() in 'ABCDEFGHIJKLMNOPQRSTUVWXYZ') != '':
        return "Calm down, I know what I'm doing!" if hey_bob.strip().endswith('?') else "Whoa, chill out!"
    elif hey_bob.strip().endswith('?'):
        return "Sure."
    else:
        return "Whatever."

File: elixir/bob/lib/bob.ex
defmodule Bob do
  def hey(input) do
    input = String.trim(input)

    uppercase =
      Enum.all?(
        Enum.filter(String.graphemes(input), &String.match?(&1, ~r/^\p{L}$/u)),
        &String.match?(&1, ~r/^\p{Lu}$/u)
      )

    has_letters = Enum.any?(String.graphemes(input), &String.match?(&1, ~r/^\p{L}$/u))
    question = String.ends_with?(input, "?")

    cond do
      String.length(input) == 0 ->
        "Fine. Be that way!"

      has_letters && uppercase && question ->
        "Calm down, I know what I'm doing!"

      question ->
        "Sure."

      has_letters && uppercase ->
        "Whoa, chill out!"

      true ->
        "Whatever."
    end
  end
end
